jsk_pcl_ros/src/hinted_plane_detector_nodelet.cpp

```diff
@@ -97,24 +97,40 @@ namespace jsk_pcl_ros {
 
   void HintedPlaneDetector::subscribe()
   {
-    sub_cloud_.subscribe(*pnh_, "input", 1);
-    sub_hint_cloud_.subscribe(*pnh_, "input/hint/cloud", 1);
-    sync_ = boost::make_shared<message_filters::Synchronizer<SyncPolicy> >(100);
-    sync_->connectInput(sub_cloud_, sub_hint_cloud_);
-    sync_->registerCallback(boost::bind(&HintedPlaneDetector::detect,
-                                        this, _1, _2));
+    if (synchronize_) {
+      sub_cloud_.subscribe(*pnh_, "input", 1);
+      sub_hint_cloud_.subscribe(*pnh_, "input/hint/cloud", 1);
+      sync_ = boost::make_shared<message_filters::Synchronizer<SyncPolicy> >(100);
```

Comment:
I think you need to add `sync_.reset()` in the destructor of the `HintedPlaneDetector` class

Reply:
Thanks for the review! Could you explain the rationale? The `sync_` exists in the original code and I don't know why we need to add it to desctructor by this change...

Reply:
Oh, I'm sorry... I overlooked that the original code also has `sync_.reset` ...